frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/ResetPasswordActivity.java

```diff
@@ -0,0 +1,329 @@
+                final int what = msg.what;
+                switch(what) {
+                    case VCodeSuccess:
+                        Toast.makeText(getApplicationContext(), "Email sent to " + email, Toast.LENGTH_SHORT).show();
+                        BConfirm.setText("Confirm");
+                        vCodeButtonClicked = true;
+                        hint.setText("Please fill in the rest and click 'Confirm'");
+                        Log.i(TAG, "Successfully sent VCode");
+                        break;
+                    case ResetSuccess:
+                        Toast.makeText(getApplicationContext(), "Password Reset Successful", Toast.LENGTH_SHORT).show();
+                        finish();
+                        break;
+                }
+            }
+        });
+        IntentFilter filter = new IntentFilter();
+        filter.addAction("vCodeSuccessful");
+        filter.addAction("ResetSuccessful");
+        registerReceiver(resetPwdReceiver, filter);
+
+        BConfirm.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!vCodeButtonClicked) {
+                    // get email from UI
+                    email = String.valueOf(emailField.getText());
+                    type = "GetVCode";
+                    vCodeBundle[0] = email;
+                    vCodeBundle[1] = type;
+
+                    // send email to server to get vCode
+                    new ResetPwdWorkingThread(getApplicationContext()).execute(vCodeBundle);
+                } else {
+                    // send to server with vCode
+                    vCode = String.valueOf(vCodeField.getText());
+                    newPwd = String.valueOf(newPwdField.getText());
+                    if (vCode.equals("")) {
+                        Toast.makeText(getApplicationContext(), "Verification code can't be empty", Toast.LENGTH_SHORT).show();
+                    } else if (newPwd.equals("")) {
+                        Toast.makeText(getApplicationContext(), "New password can't be empty", Toast.LENGTH_SHORT).show();
+                    } else {
+                        resetBundle[0] = vCode;
+                        resetBundle[1] = "Reset";
+                        resetBundle[2] = newPwd;
+
+                        new ResetPwdWorkingThread(getApplicationContext()).execute(resetBundle);
+                    }
+
+                }
+            }
+        });
+
+        BBack.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        unregisterReceiver(resetPwdReceiver);
+    }
+}
+
+class ResetPwdWorkingThread extends AsyncTask<String, Void, Void> {
+    String vCodeUrl = "http://128.199.231.190:8080/api/users/resetPassword";
```

Comment:
Actually we should make it consistent to use Singleton instance of httpbuilder, easier to maintain.